[Doc]Add documents for feature LendingLimit (#1751)
kubernetes-sigs/kueue: site/content/en/docs/concepts/cluster_queue.md

ClusterQueues in the cohort. So for the yamls listed above, `team-b-cq` can
borrow `12+9` CPUs.

### LendingLimit

To limit the amount of resources that a ClusterQueue can lend to others,
you can set the `.spec.resourcesGroup[*].flavors[*].resource[*].lendingLimit`
[quantity](https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/quantity/) field.

> #### Enabling feature LendingLimit

Reviewer: ```suggestion
```

I don't think this section level would be useful.

Author: Yes. I was trying to make it level 4 title, but that doesn't seem to work.
removed.

> 
> LendingLimit is an `Alpha` feature disabled by default, check the [Change the feature gates configuration](/docs/installation/#change-the-feature-gates-configuration) section of the [Installation](/docs/installation/) for details.
> 
> > _Available in Kueue v0.6.0 and later_
> 

Reviewer: ```suggestion
{{% alert title="Warning" color="warning" %}}
_Available in Kueue v0.6.0 and later_
`LendingLimit` is an `Alpha` feature disabled by default.
You can enable it by setting the `LendingLimit` feature gate. Check the [Installation](/docs/installation/#change-the-feature-gates-configuration) guide for details on feature gate configuration.
{{% alert title="Warning" color="warning" %}}
```

Author: Done.

As an example, assume you created the following two ClusterQueues:

```yaml
apiVersion: kueue.x-k8s.io/v1beta1
kind: ClusterQueue
metadata:
  name: "team-a-cq"
spec:
  namespaceSelector: {} # match all.
  cohort: "team-ab"
  resourceGroups:
  - coveredResources: ["cpu"]
    flavors:
    - name: "default-flavor"
      resources:
      - name: "cpu"
        nominalQuota: 9
```

```yaml
apiVersion: kueue.x-k8s.io/v1beta1
kind: ClusterQueue
metadata:
  name: "team-b-cq"
spec:
  namespaceSelector: {} # match all.
  cohort: "team-ab"
  resourceGroups:
  - coveredResources: ["cpu"]
    flavors:
    - name: "default-flavor"
      resources:
      - name: "cpu"
        nominalQuota: 12
        lendingLimit: 1
```

> In this case, because we set lendingLimit in ClusterQueue `team-b-cq`, if 
> ClusterQueue `team-b-cq` has no admitted Workloads, then ClusterQueue `team-a-cq` 
> can admit Workloads with resources adding up to `9+1=10` CPUs. 

Reviewer 2: ```suggestion
In this case, because we set the lendingLimit in ClusterQueue `team-b-cq`, 
then ClusterQueue `team-a-cq` can at most admit Workloads with quota adding 
up to `9+1=10` CPUs. 
```

> `team-a-cq`  can admit Workloads with resources adding up to `9+1=10` CPUs.

Reviewer: ```suggestion
`team-a-cq` can admit Workloads with resources adding up to `9+1=10` CPUs.
```

Author: good catch. thx!

> If, for a given flavor/resource, the `lendingLimit` field is empty or null, 
> a ClusterQueue can borrow up to the sum of nominal quotas from all the 
> ClusterQueues in the cohort. So for the yamls listed above, `team-b-cq` can 
> borrow `12+9` CPUs.

Reviewer 2: ```suggestion
If, for a given flavor/resource, the `lendingLimit` field is empty or null, 
a clusterQueue can lend out all of its resources. In the above case, 
`team-b-cq` can borrow up to `12+9=21` CPUs.
```

> If, for a given flavor/resource, the `lendingLimit` field is empty or null,
> a ClusterQueue can lend out all of its resources. So for the yamls listed above,
> `team-b-cq` can  borrow `12+9` CPUs.

Reviewer: ```suggestion
If the `lendingLimit` field is empty or null,
a ClusterQueue can lend out all of its resources. So, `team-b-cq` can borrow `12+9` CPUs.
```

Author: Done.

ClusterQueues in the cohort. So for the yamls listed above, `team-b-cq` can
borrow `12+9` CPUs.

### LendingLimit

> To limit the amount of resources that a ClusterQueue can lend to others,

Reviewer 3: ```suggestion
To limit the amount of resources that a ClusterQueue can lend in the cohort,
```

you can set the `.spec.resourcesGroup[*].flavors[*].resource[*].lendingLimit`
[quantity](https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/quantity/) field.

{{% alert title="Warning" color="warning" %}}
_Available in Kueue v0.6.0 and later_

> `LendingLimit` is an `Alpha` feature disabled by default.

Reviewer 4: ```suggestion
`LendingLimit` is an Alpha feature disabled by default.
```

You can enable it by setting the `LendingLimit` feature gate. Check the [Installation](/docs/installation/#change-the-feature-gates-configuration) guide for details on feature gate configuration.
{{% /alert %}}

As an example, assume you created the following two ClusterQueues:

```yaml
apiVersion: kueue.x-k8s.io/v1beta1
kind: ClusterQueue
metadata:
  name: "team-a-cq"
spec:
  namespaceSelector: {} # match all.
  cohort: "team-ab"
  resourceGroups:
  - coveredResources: ["cpu"]
    flavors:
    - name: "default-flavor"
      resources:
      - name: "cpu"
        nominalQuota: 9
```

```yaml
apiVersion: kueue.x-k8s.io/v1beta1
kind: ClusterQueue
metadata:
  name: "team-b-cq"
spec:
  namespaceSelector: {} # match all.
  cohort: "team-ab"
  resourceGroups:
  - coveredResources: ["cpu"]
    flavors:
    - name: "default-flavor"
      resources:
      - name: "cpu"
        nominalQuota: 12
        lendingLimit: 1
```

> In this case, because we set lendingLimit in ClusterQueue `team-b-cq`, if

Reviewer 3: nit: Some suggestion to maybe make it easier to read:
```
Here, lendingLimit=1 is set in ClusterQueue `team-b-cq`. It means that if all admitted workloads in the 
ClusterQueue `team-b-cq` have their total quota usage below the `nominalQuota` (less or equal `12-1=11` 
CPUs), then ClusterQueue `team-a-cq` can admit Workloads with resources adding up to `9+1=10` CPUs.
```

Reviewer 4: Use active voice: "you set lendingLimit=1 in ClusterQueue"

Author: Done.

ClusterQueue `team-b-cq` has no admitted Workloads using lendingLimit
(quota usage over guaranteed quota, `12-1=11` CPUs), then ClusterQueue
`team-a-cq` can admit Workloads with resources adding up to `9+1=10` CPUs.

If the `lendingLimit` field is empty or null, a ClusterQueue can lend out

> all of its resources. In this case, `team-b-cq` can  borrow `12+9` CPUs.

Reviewer 3: ```suggestion
all of its resources. In this case, `team-a-cq` can use up to `9+12` CPUs.
```
? I think we want to say from the perspective of `team-a-cq` so that it matches the description in the previous paragraph, just without limit.

Author: SGTM, done.
